Orange/tests/test_domain.py

```diff
@@ -153,41 +145,32 @@ def test_from_numpy_names(self):
                          ["Meta {:03}".format(i) for i in range(1, 101)])
 
     def test_from_numpy_dimensions(self):
-        d = Domain.from_numpy(np.zeros((1, 1)), np.zeros(5))
-        self.assertTrue(d.anonymous)
-        self.assertEqual(len(d.class_vars), 1)
-
-        d = Domain.from_numpy(np.zeros((1, 1)), np.zeros((5, 1)))
-        self.assertTrue(d.anonymous)
-        self.assertEqual(len(d.class_vars), 1)
+        for dimension in [[5], [5, 1]]:
+            d = Domain.from_numpy(np.zeros((1, 1)), np.zeros(dimension))
+            self.assertTrue(d.anonymous)
+            self.assertEqual(len(d.class_vars), 1)
 
         self.assertRaises(ValueError, Domain.from_numpy, np.zeros(2))
         self.assertRaises(ValueError, Domain.from_numpy, np.zeros((2, 2, 2)))
         self.assertRaises(ValueError, Domain.from_numpy, np.zeros((2, 2)), np.zeros((2, 2, 2)))
 
     def test_from_numpy_values(self):
-        d = Domain.from_numpy(np.zeros((1, 1)), np.arange(1, 3).reshape(2, 1))
-        self.assertTrue(d.anonymous)
-        self.assertIsInstance(d.class_var, ContinuousVariable)
-
-        d = Domain.from_numpy(np.zeros((1, 1)), np.arange(2).reshape(2, 1))
-        self.assertTrue(d.anonymous)
-        self.assertIsInstance(d.class_var, DiscreteVariable)
-        self.assertEqual(d.class_var.values, ["v{}".format(i)
-                                              for i in range(1, 3)])
-
-        d = Domain.from_numpy(np.zeros((1, 1)), np.arange(18, 23).reshape(5, 1))
-        self.assertTrue(d.anonymous)
-        self.assertIsInstance(d.class_var, ContinuousVariable)
+        for aran_min, aran_max, res_min, res_max, type in [(1, 3, 2, 1, ContinuousVariable),
```

Comment:
* `res_min` and `res_max` define shape (i.e. n_rows, n_cols); nothing to do with min/max.
* All `res_min` are equal to `aran_max - aran_min`, therefore redundant. All `res_max` are `1`.
* Don't occlude a builtin function by going with, e.g., `vartype`.